File: read-user/controllers/user_controller.py
from flask import Blueprint, jsonify
from services.user_service import get_all_users

# Create a new Blueprint for user-related routes
user_bp = Blueprint('user_bp', __name__)

@user_bp.route('/users', methods=['GET'])
def get_users():
    """
    Handle GET requests to fetch all users.
    
    Returns:
        Response: A JSON response containing the list of users or an error message.
    """
    try:
        # Fetch all users from the user service
        users = get_all_users()
        
        # Return the list of users as a JSON response with a 200 status code
        return jsonify(users), 200
    except Exception as e:
        # Return an error message as a JSON response with a 500 status code
        return jsonify({'message': 'Error fetching users', 'error': str(e)}), 500

File: create-user/src/controllers/userController.js
// Importing the userService, which contains business logic for handling user-related operations
const userService = require('../services/userService');

// Exporting the createUser controller function to handle HTTP POST requests for creating a new user
exports.createUser = async (req, res) => {
  try {
    // Attempting to create a new user using the data provided in the request body
    const user = await userService.createUser(req.body);

    // If successful, send a response with status code 201 (Created) and the newly created user as JSON
    res.status(201).json(user);
  } catch (error) {
    // If an error occurs during user creation, catch the error and send a response with status code 400 (Bad Request)
    // The response includes an error message in JSON format
    res.status(400).json({ message: error.message });
  }
};
